Code/GraphMol/FindStereo.cpp

```diff
@@ -770,6 +770,20 @@ bool updateBonds(ROMol &mol, const std::vector<unsigned int> &aranks,
       auto sinfo = detail::getStereoInfo(bond);
       ASSERT_INVARIANT(sinfo.controllingAtoms.size() == 4,
                        "bad controlling atoms size");
+
+      if ((sinfo.controllingAtoms[0] == Chirality::StereoInfo::NOATOM &&
+           sinfo.controllingAtoms[1] == Chirality::StereoInfo::NOATOM) ||
+          (sinfo.controllingAtoms[2] == Chirality::StereoInfo::NOATOM &&
+           sinfo.controllingAtoms[3] == Chirality::StereoInfo::NOATOM)) {
+        // we have a bond with no neighbors on one side, which means it must
+        // have a single implicit H on that side. Since the H is implicit,
+        // there is no way to know whether it is cis or trans.
+        ASSERT_INVARIANT(
+            sinfo.specified == StereoSpecified::Unknown,
+            "stereo bond without neighbors can only be unspecified");
```

Comment:
We generally use `ASSERT_INVARIANT` for things which should never ever happen (i.e. are due to egregious programming errors). It will either throw or assert, so the next line will never be called.
I think this should be, at most, a warning.

Reply:
I added the assert as a guard for future changes, to force us to come back to this check if we ever change the way these double bonds without neighbors are handled.

I think a warning doesn't make sense here because after #6527 `getStereoInfo()` should set stereo bonds with no neighbors (only an implicit H) on at least one one side to `sinfo.specified` to `StereoSpecified::Unknown`, so having any other value here doesn't seem possible.